airflow/www/static/js/dag/details/NotesAccordion.tsx

```diff
@@ -106,8 +160,16 @@ const NotesAccordion = ({
                     maxRows={10}
                     as={ResizeTextarea}
                     value={note}
-                    onChange={(e) => setNote(e.target.value)}
+                    onChange={(e) => {
+                      setNote(e.target.value);
+                    }}
                     data-testid="notes-input"
+                    onFocus={() => {
+                      localStorage.setItem(isInputIFocus, "true");
```

Comment:
We shouldn't use localstorage for this. What are you trying to do here?

Reply:
As mentioned [here](https://github.com/apache/airflow/pull/30950#issuecomment-1553315956), no matter what I use, the events still end up bubbling when the user is typing in the notes text box. So, for example, if the user, while typing in the notes textbox presses `shift + C` ( to get capital C ), it fires up the corresponding shortcut as well. Hence, in order to avoid triggering the shortcut, we are using this LocalStorage to identify whether the user is typing or not at any given moment [in this line](https://github.com/apache/airflow/blob/275d4a226a3059bd50b5ae6417a2d4f506c5ae1f/airflow/www/static/js/utils/useKeysPress.ts#L43). If yes, then we will ignore the event to trigger a shortcut, or if no, then otherwise. And onBlur, we are setting it to false.

Reply:
It is not the problem with the accordion but with how React handles the events and here we have a mix of flask app builder with React inserted into it, which makes the situation even more complex. Even if we got rid of the accordion, given any text box user is typing (in any part of the whole app given useKeysPress components are mounted), the events will be bubbled to the `useKeysPress`, and it will perform those actions that will not be desirable to the user (only while typing). Hence, have put this check using local storage that will completely stop it.

Reply:
Ok, we can keep it for now then.